portal/v2/src/ClusterDetailListComponents/Statistics/StatisticsComponent.tsx

```diff
@@ -0,0 +1,167 @@
+  ILineChartDataPoint,
+  ILineChartProps
+} from '@fluentui/react-charting';
+import { DefaultPalette} from '@fluentui/react/lib/Styling';
+import { IMetrics} from './StatisticsWrapper';
+import { convertToUTC } from "./GraphOptionsComponent";
+
+export function StatisticsComponent(props: {
+  metrics: IMetrics[],
+  clusterName: any,
+  duration: string,
+  height: number,
+  width: number,
+  endDate: Date,
+  fetchStatus: string,
+}) {
+  const width = props.width
+  const height = props.height
+  const [points, setPoints] = useState<ILineChartPoints[]>([])
+  const [data, setData] = useState<IChartProps>({})
+  const [spinnerVisible, setSpinnerVisible] = useState<boolean>(true)
+  const timeFormat = '%H:%M'
+
+  const colors: string[] = [
+    DefaultPalette.blue,
+    DefaultPalette.blueLight,
+    DefaultPalette.blueDark,
+    DefaultPalette.blueMid,
+    DefaultPalette.black,
+    DefaultPalette.red,
+    DefaultPalette.redDark,
+    DefaultPalette.yellow,
+    DefaultPalette.yellowDark,
+    DefaultPalette.yellowLight,
+    DefaultPalette.green,
+    DefaultPalette.greenLight,
+    DefaultPalette.greenDark,
+    DefaultPalette.purple,
+    DefaultPalette.purpleLight,
+    DefaultPalette.purpleDark,
+    DefaultPalette.orange,
+    DefaultPalette.orangeLight,
+    DefaultPalette.orangeLighter,
+    DefaultPalette.magenta,
+    DefaultPalette.magentaDark,
+    DefaultPalette.magentaLight,
+    DefaultPalette.themePrimary,
+    DefaultPalette.neutralPrimary,
+    DefaultPalette.neutralDark,
+    DefaultPalette.neutralSecondary,
+    DefaultPalette.neutralTertiary,
+    DefaultPalette.teal,
+    DefaultPalette.tealDark,
+    DefaultPalette.tealLight,
+    DefaultPalette.accent,
+    DefaultPalette.themeDarker,
+    DefaultPalette.themeDarkAlt,
+    DefaultPalette.themeDark,
+    DefaultPalette.themeLight,
+    DefaultPalette.themeLighter,
+    DefaultPalette.themeLighterAlt,
+    DefaultPalette.themePrimary,
+    DefaultPalette.themeSecondary,
+    DefaultPalette.themeTertiary
+  ]
+  
+  function StatisticsHelperComponent(): JSX.Element {   
+    useEffect(() => {
+      if (props.fetchStatus === "success") {
+        const newPoints: ILineChartPoints[] = []
+        props.metrics.forEach((metric, i) => {          
+          var dataPoints: ILineChartDataPoint[] = []
+          metric.MetricValue.forEach(metricValue => {
+            let timeStamp = new Date(metricValue.timestamp)
+            let metricsTime = convertToUTC(timeStamp)
+            var data: ILineChartDataPoint = {
+              x: metricsTime, y: metricValue.value
+            }
+            dataPoints.push(data)
+          })
+          var lineChartPoint: ILineChartPoints = {
+            legend: metric.Name,
+            data: dataPoints,
+            color: colors[i]
+          }
+          newPoints.push(lineChartPoint)
+        })
+        setPoints(newPoints);
+      }
+      (props.fetchStatus === "error") ? setSpinnerVisible(false) : setSpinnerVisible(true)
```

Comment:
Consider refactoring this so that you remove 1 if statement:
```
if (props.fetchStatus === "error") {
  ... do things
  return
}

// success actions here
```

Reply:
Yup, got it ... will refactor it.